From e7699fcd0289f2ee075e6ba216d260c49e494059 Mon Sep 17 00:00:00 2001
From: Jeff Lin <jeffulin@google.com>
Date: Mon, 1 Jul 2024 15:37:35 +0800
Subject: [PATCH] Do not use boringssl in base::UnguessableToken

patch-name: long-term-0700-Do-not-use-boringssl-in-base-UnguessableToken.patch
Change-Id: Ibf30a5a96b451f58c79619b04fb6a5646d1c6be0
---
 base/unguessable_token.cc | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/base/unguessable_token.cc b/base/unguessable_token.cc
index e996816597..f7be0d1be8 100644
--- a/base/unguessable_token.cc
+++ b/base/unguessable_token.cc
@@ -11,7 +11,6 @@
 #include "base/format_macros.h"
 #include "base/rand_util.h"
 #include "build/build_config.h"
-#include "third_party/boringssl/src/include/openssl/mem.h"
 
 namespace base {
 
@@ -55,9 +54,7 @@ std::optional<UnguessableToken> UnguessableToken::DeserializeFromString(
 }
 
 bool operator==(const UnguessableToken& lhs, const UnguessableToken& rhs) {
-  auto bytes = lhs.token_.AsBytes();
-  auto other_bytes = rhs.token_.AsBytes();
-  return CRYPTO_memcmp(bytes.data(), other_bytes.data(), bytes.size()) == 0;
+  return lhs.token_ == rhs.token_;
 }
 
 std::ostream& operator<<(std::ostream& out, const UnguessableToken& token) {
-- 
2.56.0.rc1.315.gc6ed9934b7-goog

